server/services/team/service.ts

```diff
@@ -589,6 +589,12 @@ export class TeamService extends ServiceClass implements ITeamService {
 		return this.TeamModel.findOneByName(teamName);
 	}
 
+	async getOneByNames(teamNames: string[]): Promise<ITeam|undefined> {
+		return this.TeamModel.findOne({
+			name: { $in: teamNames },
+		});
+	}
+
 	async getOneByRoomId(roomId: string): Promise<ITeam | null> {
```

Comment:
Since this function could be used in other places, I'd recommend taking the projection as a param